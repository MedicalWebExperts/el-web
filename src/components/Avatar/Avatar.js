import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from '../../theme';

const propTypes = {
  image: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

const Avatar = ({
  image, alt,
}) => (
  <div>
    <style jsx>
      {`
        img{
          border-radius: 100%;
          display: block;
          width: 100px;
          height: auto;
        }
      `}
    </style>
    <img src={image} alt={alt}/>
  </div>
);

Avatar.propTypes = propTypes;
export default withTheme(Avatar);
