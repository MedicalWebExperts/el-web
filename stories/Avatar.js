import React from 'react';
import { storiesOf } from '@storybook/react';

import { ThemeProvider, Avatar } from '../src';

export default () => {
  storiesOf('Avatar', module).add('Avatar Normal size', () => (
    <ThemeProvider>
      <Avatar image="https://placehold.it/350x350" alt="Image alt" />
    </ThemeProvider>
  ));

  storiesOf('Avatar', module).add('Avatar Small size', () => (
    <ThemeProvider>
      <Avatar image="https://placehold.it/350x350" alt="Image alt" size="small" />
    </ThemeProvider>
  ));

  storiesOf('Avatar', module).add('Avatar Large size', () => (
    <ThemeProvider>
      <Avatar image="https://placehold.it/350x350" alt="Image alt" size="large" />
    </ThemeProvider>
  ));

}
