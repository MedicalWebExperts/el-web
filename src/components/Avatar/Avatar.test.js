import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

const defaultProps = {
  image: 'https://placehold.it/350x350',
  alt: 'Avatar Test',
};

const customProps = {
  image: 'https://placehold.it/350x350',
  alt: 'Avatar Test',
  size: 'large',
};

const customStyles = {
  borderRadius: '20px',
  padding: '0',
};

describe('Component Avatar Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<Avatar {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('diferent avatar size', () => {
    const tree = renderer.create(<Avatar {...customProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using custom styles', () => {
    const tree = renderer.create(<Avatar {...defaultProps} styles={customStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
