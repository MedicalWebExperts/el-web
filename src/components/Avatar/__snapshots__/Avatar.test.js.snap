// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Avatar Snapshot diferent avatar size 1`] = `
<div
  className="jsx-1666591538"
>
  <img
    alt="Avatar Test"
    className="jsx-1666591538"
    src="https://placehold.it/350x350"
    style={Object {}}
  />
</div>
`;

exports[`Component Avatar Snapshot using custom styles 1`] = `
<div
  className="jsx-46401105"
>
  <img
    alt="Avatar Test"
    className="jsx-46401105"
    src="https://placehold.it/350x350"
    style={
      Object {
        "borderRadius": "20px",
        "padding": "0",
      }
    }
  />
</div>
`;

exports[`Component Avatar Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-46401105"
>
  <img
    alt="Avatar Test"
    className="jsx-46401105"
    src="https://placehold.it/350x350"
    style={Object {}}
  />
</div>
`;
